pkg/kratos/handlers.go

```diff
@@ -206,10 +207,56 @@ func (a *API) handleUpdateFlow(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	shouldRegenerateBackupCodes, err := a.shouldRegenerateBackupCodes(r.Context(), cookies)
+	if err != nil {
+		err = fmt.Errorf("error when checking backup codes: %v", err)
+
+		a.logger.Error(err.Error())
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
```

Comment:
Not sure if I understand correctly, so please correct me if I'm wrong:

This endpoint may be called multiple times during a user's login flow, once for the username/password, once for OTP, once for webauthn. This means that this logic will run multiple times and if I don't want to print new codes (which is not unlikely) I will have to decline this every time. 

I have not run it to see if this is the actual behavior, so I may be missing something.

Reply:
The user will only be offered to regenerate the codes if the method used to perform MFA was lookup secret and there are 3 or less codes left.
The screen is not presented multiple times (there are checks in the `shouldRegenerateBackupCodes` handler to prevent this) so the user can only decline it once.

Reply:
Ah, I missed that thanks. What if we have more than 2fa though? Is this possible to happen (passkey as 3rd fa or something like that)

Reply:
Passkey can only be used as the first factor (this is restricted by kratos settings). No matter how many factors are there, this screen will only be shown once when a backup code is used.